components/mcp3008rst

```diff
@@ -0,0 +1,70 @@
+MCP3008 I/O Expander
+=====================
+
+.. seo::
+    :description: Instructions for setting up MCP3008 10 Bit Analog to Digital Converter in ESPHome.
+    :image: mcp3008.png
+
+The Microchip Technology Inc. MCP3008
+devices are successive approximation 10-bit Analogto-Digital (A/D) converters with on-board sample and
+hold circuitry
+
+MCP3008
+--------
+
+The MCP3008 component allows you to use MCP3008 8-Channel 10-Bit A/D Converter
+(`datasheet <http://ww1.microchip.com/downloads/en/DeviceDoc/21295d.pdf>`__,
+`Adafruit <https://www.adafruit.com/product/856>`__) in ESPHome.
+It uses the :ref:`SPI Bus <spi>` for communication.
+
+Once configured, you can use any of the 8 pins as
+sensors for your projects. 
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    MCP3008:
+      - id: 'MCP3008_hub'
+        cs_pin: D8
+
+    # Individual outputs
+    sensor:
+      - platform: mcp3008
+        update_interval: 1s
+        mcp3008: 'mcp3008_hub'
+        id: freezer_temp_source
+        name: "Freezer Temp"
+        number: 0
+      - platform: mcp3008
+        update_interval: 1s
+        mcp3008: 'mcp3008_hub'
+        name: "FridgeTmep"
+        number: 1
+      - platform: resistance
+        id: resistance_sensor
+        sensor: freezer_temp_source
+        configuration: DOWNSTREAM
+        resistor: 10kOhm
+        name: Resistance Sensor
```

Comment:
hide name here, so resistance sensor does not go to HA

Reply:
talk about confusion. I guess i was excited to send this PR in. I typically dont contribute but am trying to be a better open source developer. Also building a home thermostat and need some components and love esphome so... some PRs from me. 

 # Example configuration entry
    MCP3008:
      - id: 'MCP3008_hub'
        cs_pin: D8

    # Example config of sensors. 
    # This is a NTCB3950 10K thermocoupler attached to pin 0
    # of the MCP3008 with a 10K resistor as a voltage divider. 
    # See `resistance` and `ntc` platorms for other options
    sensor:
      - platform: mcp3008             # Attached to pin 0 of the MCP3008. The result will be a reading from 0-1023
        update_interval: 1s
        mcp3008: 'mcp3008_hub'
        id: freezer_temp_source       
        number: 0                     # pin number
      - platform: resistance
        id: freezer_resistance_sensor 
        sensor: freezer_temp_source
        configuration: DOWNSTREAM
        resistor: 10kOhm
        name: Resistance Sensor
      - platform: ntc
        id: freezer_temp
        sensor: freezer_resistance_sensor
        calibration:
          b_constant: 3950
          reference_temperature: 25°C
          reference_resistance: 10kOhm
        name: Freezer Temperature